src/controller.js

```diff
@@ -283,8 +286,11 @@ TreemapController.overrides = {
         label(item) {
           const dataset = item.dataset;
           const dataItem = dataset.data[item.dataIndex];
-          const label = dataItem.g || dataset.label;
-          return (label ? label + ': ' : '') + dataItem.v;
+          if (dataItem.g) {
+            return dataItem.g + ': ' + dataItem.v;
+          }
+          const value = item.element.options.formatter;
+          return value || dataItem.v;
```

Comment:
I'd leave this unchanged if you agree with my suggestions.

Reply:
Done! I haven't committed yet, waiting feedbacks for above item